shared/src/main/scala/effekt/Namer.scala

```diff
@@ -45,6 +45,10 @@ class Namer extends Phase[Module, Module] { namer =>
     val imports = mod.decl.imports map {
       case im @ source.Import(path) => Context.at(im) {
         val modImport = Context.moduleOf(path)
+        // compare path with modImport.path to ensure correct handling of case sensitivity
+        if (modImport.decl.path != path) {
+          Context.abort(s"Could find an import, but has wrong module path.\nPath declared in the module: ${modImport.decl.path}")
```

Comment:
Maybe we can be even more specific and say "Path declared in the *imported* module:".
WDYT?